ts/src/main/scala/ch/epfl/ts/example/FullMarketSimulation.scala

```diff
@@ -0,0 +1,148 @@
+    initProducersAndConsuments()
+
+    //trader
+    val parameters = new StrategyParameters(
+      MadTrader.INITIAL_FUNDS -> WalletParameter(Map(Currency.CHF -> 1000.0)),
+      MadTrader.INTERVAL -> new TimeParameter(1 seconds),
+      MadTrader.ORDER_VOLUME -> NaturalNumberParameter(10),
+      MadTrader.CURRENCY_PAIR -> new CurrencyPairParameter(Currency.USD, Currency.CHF))
+
+    val tId = 15L
+
+    val trader = MadTrader.getInstance(tId, parameters, "OneMadTrader")
+    addConsument(classOf[Quote], trader)
+    val broker = builder.createRef(Props(classOf[StandardBroker]), "Broker")
+    addConsument(classOf[Quote], broker)
+
+    trader->(broker, classOf[Register])
+    trader->(broker, classOf[FundWallet])
+    connectAllOrders(trader, broker)
+
+    //fetcher
+    val useLiveData = false
+    val symbol = (Currency.USD, Currency.CHF)
+    val fxQuoteFetcher = createFetcher(useLiveData, builder, symbol)
+    addProducer(classOf[Quote], fxQuoteFetcher)
+
+    //hybrid market
+    val fetcherRules = new FxMarketRulesWrapper()
+    val simulationRules = new SimulationMarketRulesWrapper()
+    val marketForexId = MarketNames.FOREX_ID
+    val forexMarket = builder.createRef(Props(classOf[HybridMarketSimulator], marketForexId, fetcherRules, simulationRules), MarketNames.FOREX_NAME)
+
+    addConsument(classOf[Quote], forexMarket)
+    addProducer(classOf[Quote], forexMarket)
+    connectAllOrders(broker, forexMarket)
+    forexMarket->(broker, classOf[ExecutedBidOrder], classOf[ExecutedAskOrder])
+
+    //Backloop
+    val dummyPersistor = new DummyPersistor()
+    val backloop = builder.createRef(Props(classOf[BackLoop], marketForexId, dummyPersistor), "backloop")
+
+    forexMarket->(backloop, classOf[Transaction])
+    addProducer(classOf[Quote], backloop)
+
+    connectProducersWithConsuments()
+
+    builder.start
+    scheduleChange(fxQuoteFetcher, forexMarket)
+  }
+
+  def scheduleChange(quoteFetcher: ComponentRef, market: ComponentRef) = {
+    val timer = new Timer()
+    class StopFetcher extends java.util.TimerTask {
+      def run() {
+        quoteFetcher.ar ! StopSignal
+        market.ar ! 'ChangeMarketRules
+      }
+    }
+    timer.schedule(new StopFetcher, 10 * 1000)
+  }
+
+  def initProducersAndConsuments() = {
+    val messageClasses = List(classOf[Quote], classOf[Transaction])
+    for(msg <- messageClasses) {
+      producers = producers + (msg -> List())
+      consuments = consuments + (msg -> List())
+    }
+  }
+
+  def addProducer(msg: Class[_], component: ComponentRef) =
+    producers = addEntity(msg, component, producers)
+
+  def addConsument(msg: Class[_], component: ComponentRef) =
+    consuments = addEntity(msg, component, consuments)
+
+  def addEntity(msg: Class[_], component: ComponentRef, mapping: Map[Class[_], List[ComponentRef]]) =
+    mapping + (msg -> (component :: mapping.getOrElse(msg, List())))
+
+  def connectProducersWithConsuments() = {
```

Comment:
All these functions are really nice but unrelated to full market simulation, could they be refactored to a helper class superclass and used in all our example classes?